Lib/test/test_math.py

```diff
@@ -2503,6 +2503,50 @@ def test_input_exceptions(self):
         self.assertRaises(TypeError, math.atan2, 1.0)
         self.assertRaises(TypeError, math.atan2, 1.0, 2.0, 3.0)
 
+    def test_exception_messages(self):
+        x = -1.1
+
+        with self.assertRaises(ValueError,
+                               msg=f"expected a nonnegative input, got {x}"):
+            math.sqrt(x)
+
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log(x)
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log(123, x)
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log2(x)
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log2(x)
+
+        x = decimal.Decimal(x)
+
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x!r}"):
+            math.log(x)
+
+        x = fractions.Fraction(1, 10**400)
+
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {float(x)!r}"):
+            math.log(x)
+
+        x = -2**1000
```

Comment:
This can be converted to float. Maybe test with small negative integer and with large negative integer that cannot be converted to float (this should not raise an OverfowError)?

Reply:
Sure, but the point is just to test that appropriate exception raised for some code flow.  Any integer trigger that path - and any negative integer fits our need here.  I'll convert to -123.